app/controllers/webauthn_verifications_controller.rb

```diff
@@ -1,13 +1,17 @@
+# This controller is for the user interface Webauthn challenge after a user follows a link generated
+# by the APIv1 WebauthnVerificationsController (controllers/api/v1/webauthn_verifications_controller).
 class WebauthnVerificationsController < ApplicationController
-  before_action :set_user
+  before_action :set_verification, :set_user
+  after_action :expire_webauthn_verification, only: :authenticate
```

Comment:
I think it makes sense to expire after a "successful" authenticate (ie. before rendering the success page with the otp). There are some cases where `authenticate` results in an error ([relevant code](https://github.com/rubygems/rubygems.org/blob/webauthn-cli/app/controllers/webauthn_verifications_controller.rb#L27-L30)). I don't think the token should expire in those error cases since the user would want to authenticate again.

Reply:
Makes sense. I can change it.

Reply:
I looked into it a bit. My reading of [the relevant Rails docs](https://guides.rubyonrails.org/action_controller_overview.html#after-filters-and-around-filters) is that it won't run when there's an exception raised:

> Please note that "after" filters are executed only after a successful action, but not when an exception is raised in the request cycle.

But upon testing this _doesn't_ seem to be the behaviour. It may be that the docs mean "if the exception is raised _and not rescued_", ie that Rails handles it instead of the controller code. Either way the necessary change is to move the call into `#authenticate`.